src/Microsoft.Repl/Parsing/CoreParser.cs

```diff
@@ -14,6 +14,9 @@ public ICoreParseResult Parse(string commandText, int caretPosition)
             commandText = commandText ?? throw new ArgumentNullException(nameof(commandText));
 
             List<string> sections = commandText.Split(' ').ToList();
+            // We can't use StringSplitOptions.RemoveEmptyEntries because it
+            // is more aggressive than we need, so we need to do it ourselves.
+            RemoveEmptyEntries(sections);
```

Comment:
Using `StringSplitOptions.RemoveEmptyEntries` will remove empty entries at the end of the string and we have a good bit of code and tests that rely on empty entries at the end and provide specific error messages for those scenarios. To keep those intact, I rolled my own.